tests/UnifiedSpecTests/RunOnRequirement.php

```diff
@@ -73,6 +83,13 @@ public function isSatisfied(string $serverVersion, string $topology) : bool
             return false;
         }
 
+        if (isset($this->serverParameters)) {
+            $constraint = new Matches($this->serverParameters, null, true, false);
```

Comment:
Since we're allowing extra root-level keys and disabling operators, I suppose the main benefit to use Matches here instead of iterating on `$this->serverParameters` (and ensuring each value in the spec test equals (`==`) something in `$serverParameters`) is the preparation logic in `Matches::prepare()`? SGTM, just curious what your reasons were for using this here.

Reply:
My original idea was to allow for using matching logic when comparing parameters, but Jeff indicated that this would be difficult in some drivers (e.g. Java), hence the omission. Since it's easier here to use that logic that's what I've gone with. Theoretically, it would allow us to use the following requirement:
```json
{
    "requireApiVersion": { "$$unsetOrMatches": false }
}
```

Drivers would then call `getParameters` once and compare the requirement against the document and run the test if this passes.

Reply:
That would surely require some adjustments to https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#runonrequirement. I see it currently refers to flexible numeric comparisons but nothing else.

Anyway, this is fine as-is. Thanks for the context.